### ISSUE TITLE
Make the app good enough to pass Android Study Jam

### ISSUE THREAD

**Issue by gmacario** [gmacario/colorcode#2]
Review the [Final Project Rubric: Android For Beginners](https://docs.google.com/document/d/1zJdkW0pFZO201KlR7JTnNBCeG3afjHRMtffCOgzqLZg/pub)

**TIP**: If the code Does Not Meet Specifications, learn from Yoda:

> "Use the (open) source, Luke!"

```
$ git clone git@github.com:gmacario/colorcode.git
$ cd colorcode
$ git remote add yoda https://github.com/WillyShakes/UdooWtf
$ git fetch --all
$ git merge yoda/master
...
$ git push
```

**Comment by gmacario** [comment]
Reviewing master branch against The Rubric

Idea Definition
- [x] Does the Idea document define the idea for the final app clearly? **Meets Specifications** - The app idea and target user population are clearly defined. The app idea leverages the following prompt: “5 ways to _____”

Solution Design and README
- [x] Does the submission include images of the app design? **Meets Specifications** - The student submission includes images of the app design.
- [x] Is the README file present and complete, clearly providing all the necessary instructions for running the program? **Meets Specifications** - The README file provides instructions on how to setup the Android project and run the app. This instruction will be used by Udacity to review your project.

Code Functionality
- [ ] Does the code fully and satisfactorily implement the idea listed in the idea definition? **Meets Specifications** - The code meets all the requirements listed in the idea definition. 
- [x] Does the code run without errors? **Meets Specifications** - The code runs without errors.
- [x] Does the code feature the required Views/ViewGroups? **Meets Specifications** - The app includes at least four of the following Views: TextView, ImageView, Button, Checkbox, EditText, LinearLayout, RelativeLayout, ScrollView. If applicable, the app uses nested ViewGroups to reduce the complexity of the layout. (See https://github.com/gmacario/colorcode/issues/9)
- [x] Does the app follow best practices for building UI? **Meets Specifications** - The code adheres to all of the following best practices:
  - Text sizes are defined in sp
  - Lengths are defined in dp
  - Padding and margin is used appropriately, such that the views are not crammed up against each other.
- [ ] Is the app interactive? **Meets Specifications** - All Button(s) have onClick events defined and function appropriately.
- [ ] Does the app demonstrate proper use of variables in his/her code? **Meets Specifications** - Variables are scoped appropriately
- [ ] Does the app use methods appropriately? **Meets Specifications** - The methods in the Java files :
  - Are defined and called
  - Do what their names suggest
  - Have appropriate input parameters and return data types
- [ ] Are if/else statement(s) used in the code? **Meets Specifications** - The app contains at least one if/else statement

Code Readability
- [x] Are comments used effectively to explain different parts of the code? **Meets Specifications** - Code is easily readable such that a fellow programmer can understand the purpose of the app.
- [x] Are variable and method names and resource IDs descriptive? **Meets Specifications** - All variables, methods, and resource IDs are descriptively named such that another developer reading the code can easily understand their function.
- [ ] Is the code properly formatted? **Meets Specifications** - The code is properly formatted i.e. there are no unnecessary blank lines;  there are no unused variables or methods; there is no commented out code. The code also has proper indentation when defining variables and methods.

**Comment by WillyShakes** [comment]
see #34 and #33

**Comment by gmacario** [comment]
Hi @WillyShakes,

Thanks a lot for your help.

I merged PR #33 and #34, now the issue may be closed.
